[DCP] Provides default AsyncStager

LucasLLC opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #124944
* __->__ #124939
* #122965

Differential Revision: [D56575987](https://our.internmc.facebook.com/intern/diff/D56575987/)

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @chauhang @d4l3k

LucasLLC committed:
[DCP] Provides default AsyncStager



Differential Revision: [D56575987](https://our.internmc.facebook.com/intern/diff/D56575987/)

[ghstack-poisoned]

LucasLLC committed:
Update on "[DCP] Provides default AsyncStager"


Differential Revision: [D56575987](https://our.internmc.facebook.com/intern/diff/D56575987/)

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 chauhang d4l3k

[ghstack-poisoned]

LucasLLC committed:
Update on "[DCP] Provides default AsyncStager"


Differential Revision: [D56575987](https://our.internmc.facebook.com/intern/diff/D56575987/)

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 chauhang d4l3k

[ghstack-poisoned]

LucasLLC committed:
Update on "[DCP] Provides default AsyncStager"


Differential Revision: [D56575987](https://our.internmc.facebook.com/intern/diff/D56575987/)

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 chauhang d4l3k

[ghstack-poisoned]

LucasLLC committed:
Update on "[DCP] Provides default AsyncStager"


Differential Revision: [D56575987](https://our.internmc.facebook.com/intern/diff/D56575987/)

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 chauhang d4l3k

[ghstack-poisoned]

fegin commented:
The PR looks good to me. The BC linter may be flaky due to the inheritance change but it is good to double check. There is also a linter issue.

LucasLLC committed:
Update on "[DCP] Provides default AsyncStager"


Differential Revision: [D56575987](https://our.internmc.facebook.com/intern/diff/D56575987/)

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 chauhang d4l3k

[ghstack-poisoned]

LucasLLC committed:
Update on "[DCP] Provides default AsyncStager"


Differential Revision: [D56575987](https://our.internmc.facebook.com/intern/diff/D56575987/)

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 chauhang d4l3k

[ghstack-poisoned]

LucasLLC commented:
@pytorchbot merge

LucasLLC commented:
@pytorchbot merge
